src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -153,14 +153,13 @@ private ICollection<Guid> TestCases(IReadOnlyList<Mutant> mutants, Dictionary<in
             return testCases;
         }
 
-        private TestRunResult BuildTestRunResult(IRunResults testResults, int expectedTests, int totalCountOfTests,
-            bool compressAll = true)
+        private TestRunResult BuildTestRunResult(IRunResults testResults, int expectedTests, int totalCountOfTests, bool compressAll = true)
         {
             var resultAsArray = testResults.TestResults.ToArray();
             var testCases = resultAsArray.Select(t => t.TestCase.Id).ToHashSet();
             var ranTestsCount = testCases.Count;
-            var timeout = !_aborted && ranTestsCount < expectedTests;
-            var ranTests = compressAll && ranTestsCount >= totalCountOfTests ? (ITestGuids)TestGuidsList.EveryTest() : new WrappedGuidsEnumeration(testCases);
+            var timeout = !_cancelled && ranTestsCount < expectedTests;
+            var ranTests = compressAll && totalCountOfTests>0 && ranTestsCount >= totalCountOfTests ? (ITestGuids)TestGuidsList.EveryTest() : new WrappedGuidsEnumeration(testCases);
```

Comment:
So thinking out loud here - I guess at some point it could be useful to just write down all the known scenarios here, like:
- 1 source project x 1 test project
- 1 source project x many test projects
- many source projects x one test project
- many source projects x many test projects
- empty source projects
- empty test projects
- ... 

... and then put this in code (read - design patterns, interfaces, whatever). 

It seems like sooner or later in the flow most of these scenarios need special treatment hence I am under the impression that the related code is gradually turning into a bunch of if/else hacks and becoming a bit hard to follow.

My impression can be wrong, that's basically a general reminder to keep an eye on the design in the light of newly appearing edge cases and scenarious.

Reply:
Well, I put my comment here because these lines just feel like we are starting to take too many circumstances into account, but generally it's unrelated :) 

But yeah I guess I am overthinking here a bit, eventually we might also just end up with some kind of wrapper on top of VsTest which would effectively carry the semantics of "this is how VsTest would work if it was designed properly and implemented without bugs", basically cutting corners and streamlining edge cases.

Reply:
Comment has been added.
VsTestRunner more or less acts as this VsTest wrapper. Maybe a better design will emerge or be found in the coming months. For my own part, I feel like I have a done a lot of redesign in the past year, and I want to focus on fixing things for the time being 😄

Reply:
Sure, that's fine. I am just sharing thoughts here, no action required :D